crates/anvil/src/lib.rs

```diff
@@ -334,44 +334,54 @@ impl NodeHandle {
         ProviderBuilder::new(&self.config.get_ipc_path()?).build().ok()
     }
 
-    /// Signer accounts that can sign messages/transactions from the EVM node
+    /// Signer accounts that can sign messages/transactions from the EVM node.
     pub fn dev_accounts(&self) -> impl Iterator<Item = Address> + '_ {
         self.config.signer_accounts.iter().map(|wallet| wallet.address())
     }
 
-    /// Signer accounts that can sign messages/transactions from the EVM node
+    /// Signer accounts that can sign messages/transactions from the EVM node.
     pub fn dev_wallets(&self) -> impl Iterator<Item = PrivateKeySigner> + '_ {
         self.config.signer_accounts.iter().cloned()
     }
 
-    /// Accounts that will be initialised with `genesis_balance` in the genesis block
+    /// Accounts that will be initialised with `genesis_balance` in the genesis block.
     pub fn genesis_accounts(&self) -> impl Iterator<Item = Address> + '_ {
         self.config.genesis_accounts.iter().map(|w| w.address())
     }
 
-    /// Native token balance of every genesis account in the genesis block
+    /// Native token balance of every genesis account in the genesis block.
     pub fn genesis_balance(&self) -> U256 {
         self.config.genesis_balance
     }
 
-    /// Default gas price for all txs
+    /// Default gas price for all txs.
     pub fn gas_price(&self) -> u128 {
         self.config.get_gas_price()
     }
 
-    /// Returns the shutdown signal
+    /// Returns the shutdown signal.
     pub fn shutdown_signal(&self) -> &Option<Signal> {
         &self._signal
     }
 
-    /// Returns mutable access to the shutdown signal
+    /// Returns mutable access to the shutdown signal.
     ///
-    /// This can be used to extract the Signal
+    /// This can be used to extract the Signal.
     pub fn shutdown_signal_mut(&mut self) -> &mut Option<Signal> {
         &mut self._signal
     }
 
-    /// Returns the task manager that can be used to spawn new tasks
+    /// Fires shutdown signal.
+    ///
+    /// Called by long-running tests to make sure anvil instance is terminated.
+    #[track_caller]
+    pub fn fire_shutdown_signal(&mut self) {
```

Comment:
Can this go into an `impl Drop`?

Reply:
yes, will do and make some tests

Reply:
hm, weird, I added in drop

```rust
+impl Drop for NodeHandle {
+    fn drop(&mut self) {
+        if let Some(signal) = self._signal.take() {
+            signal.fire().unwrap()
+        }
+    }
+}
```

and then got again a failure like
```
failed to spawn node: Os { code: 98, kind: AddrInUse, message: "Address already in use" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

failures:
    proof::test_account_proof
```

going to make some more tests to see if consistently happens

Reply:
added in https://github.com/foundry-rs/foundry/pull/8947/commits/277d62c22a3bcc2484825c3d633517d7df8ac0ca ptal